src/Lecture2.hs

```diff
@@ -51,8 +55,14 @@ zero, you can stop calculating product and return 0 immediately.
 >>> lazyProduct [4, 3, 7]
 84
 -}
+
+
 lazyProduct :: [Int] -> Int
-lazyProduct = error "TODO"
+lazyProduct list = go 1 list
+  where
+    go acc [] = acc
+    go acc (x:xs) = if x == 0 then go 0 [] else go ( acc * x )  xs
```

Comment:
You can use pattern-matching here to write this in a slightly more elegant way 🙂 
```suggestion
    go _ (0:_) = 0
    go acc (x:xs) = go ( acc * x )  xs
```

Reply:
This is a great implementation !